lib/arjdbc/vertica/adapter.rb

```diff
@@ -67,12 +87,32 @@ module Vertica
     }
 
     TIMESTAMP_COLUMNS = [
-      "created_at",
-      "created_on",
-      "updated_at",
-      "updated_on"
+        "created_at",
+        "created_on",
+        "updated_at",
+        "updated_on"
     ]
 
+    def columns(table_name, name = nil)
+      sql = "SELECT * from V_CATALOG.COLUMNS WHERE table_name = '#{table_name}';"
+      raw_columns = execute(sql, name || "SCHEMA")
+
+      # Vertica doesn't use the type name of bigint since all ints are 8-byte, so when the type is int use bigint instead
+
+      columns = raw_columns.map do |raw_column|
+        ::ActiveRecord::ConnectionAdapters::VerticaColumn.new(
+            raw_column['column_name'],
+            raw_column['column_default'],
+            raw_column['data_type_id'],
+            raw_column['table_name'],
+            raw_column['data_type'] == 'int' ? 'bigint' : raw_column['data_type'],
+            fetch_type_metadata(raw_column['data_type'] == 'int' ? 'bigint' : raw_column['data_type']),
```

Comment:
What is the difference between this line and the one above it?

Reply:
line 108 passes the name of the data type into the constructor, 

line 109 passes a `SqlTypeMetadata` object into the constructor  
https://github.com/rails/rails/blob/6-1-stable/activerecord/lib/active_record/connection_adapters/sql_type_metadata.rb

which contains info about limit/precision and other things

Reply:
this did make me remember a bug i found related to the VerticaColumn constructor that i need fix still